lib/edge_builder/repositories/vehicle_repo.ex

```diff
@@ -0,0 +1,47 @@
+defmodule EdgeBuilder.Repositories.VehicleRepo do
+  alias EdgeBuilder.Repo
+  alias EdgeBuilder.Models.Vehicle
+  import Ecto.Query, only: [from: 2]
+
+  def all(page \\ 0) do
+    Repo.paginate(
+      (from v in Vehicle,
+      order_by: [desc: v.inserted_at]),
+      page: page)
+    |> callbacks_paginated()
+  end
+
+  def all_for_user(user_id) do
+    Repo.all(
+      from v in Vehicle,
+      where: v.user_id == ^user_id,
+      order_by: [desc: v.inserted_at])
+    |> Enum.map(&callbacks/1)
+  end
+
+  def recent do
+    Repo.all(
+      from v in Vehicle,
+      order_by: [desc: v.inserted_at],
+      limit: 5)
+    |> Enum.map(&callbacks/1)
+  end
+
+  def full_vehicle(permalink) do
```

Comment:
I think this is the only public method on this module which is actually called. The others above it are all dead code AFAICT. I'd suggest either getting rid of those methods or updating the appropriate places in the controllers where they should be called